fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentRegistry.java

```diff
@@ -43,6 +43,21 @@ public final class AttachmentRegistry {
 	private AttachmentRegistry() {
 	}
 
+	/**
+	 * Creates <i>and registers</i> an attachment, configuring the builder used underneath.
+	 *
+	 * @param id  the identifier of this attachment
+	 * @param <A> the type of attached data
+	 * @return the registered {@link AttachmentType} instance
+	 */
+	public static <A> AttachmentType<A> create(Identifier id, Consumer<Builder<A>> consumer) {
```

Comment:
Since it's intended as a replacement for `builder()`, this pattern should be explicitly documented in the javadoc above.

Reply:
Since `builder()` doesn't work as intended anyway, and this *is* an experimental module, would removing or deprecating the existing `builder()` method make sense?

Reply:
Removing is definitely too much. Honestly, I knew of the issue already, I just thought passing the type parameters was alright. There's absolutely no need to make a breaking change, considering the existing method works, even if it's a bit clunky.

Deprecation sounds a bit more reasonable, but I think @modmuss50 knows the policy better on that front.

Reply:
Deprecating the old methods (without forRemoval) seems fine, there is no benefit to removing them.